src/Imagine.php

```diff
@@ -97,14 +97,18 @@ private function doLoad(string $data, MetadataBag $metadata): Image
         }
 
         $internalErrors = libxml_use_internal_errors(true);
-        $disableEntities = libxml_disable_entity_loader();
         libxml_clear_errors();
 
+        $disableEntities = LIBXML_VERSION < 20900 ? libxml_disable_entity_loader() : true;
```

Comment:
It seems that `$disableEntities` is only used inside the if condition in line 110. You may want to move the code there (or even use `libxml_disable_entity_loader(libxml_disable_entity_loader())`).

Reply:
The idea is to disable the entity loader before `$document->loadXML()` and revert the setting afterwards.

My first version looked like this:

```suggestion
        if (LIBXML_VERSION < 20900) {
            $disableEntities = libxml_disable_entity_loader();
        }
```

But phpstan or psalm had a problem with that (because it thought the variable might be undefined on line 110).

Reply:
That is actually a bug in PHPStan/Psalm, however, I doubt that it can be fixed there. You could do:

```php
$disableEntities = null;

// Disable the entity loader
if (LIBXML_VERSION < 20900) {
    $disableEntities = libxml_disable_entity_loader();
}

// …

// Reset the entity loader
if (LIBXML_VERSION < 20900) {
    libxml_disable_entity_loader($disableEntities);
}
```

But you might as well leave it as is. 😄

Reply:
Done in 1aec39d66e4b8bff995fdaf88f8faa0479fd3e17